File: GXpert/GXpert.Web/Modules/Masters/UserAcademic/UserAcademicColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.UserAcademic")]
[BasedOnRow(typeof(UserAcademicRow), CheckNames = true)]
public class UserAcademicColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string StudentPrn { get; set; }
    public string AcademicYearName { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public bool IsVerify { get; set; }
    public int VerifyBy { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/UserAcademicColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UserAcademicRow } from "./UserAcademicRow";

export interface UserAcademicColumns {
    Id: Column<UserAcademicRow>;
    StudentPrn: Column<UserAcademicRow>;
    AcademicYearName: Column<UserAcademicRow>;
    Title: Column<UserAcademicRow>;
    Description: Column<UserAcademicRow>;
    IsVerify: Column<UserAcademicRow>;
    VerifyBy: Column<UserAcademicRow>;
}

export class UserAcademicColumns extends ColumnsBase<UserAcademicRow> {
    static readonly columnsKey = 'Masters.UserAcademic';
    static readonly Fields = fieldsProxy<UserAcademicColumns>();
}